contracts/truefi2/TrueFiPool2.sol

```diff
@@ -391,6 +398,20 @@ contract TrueFiPool2 is ITrueFiPool2, ERC20, Claimable {
         emit Collected(beneficiary, amount);
     }
 
+    /**
+     * @dev Switches current strategy to a new strategy
+     * @param newStrategy strategy to switch to
+     */
+    function switchStrategy(ITrueStrategy newStrategy) external onlyOwner {
+        require(strategy != newStrategy, "TrueFiPool: cannot switch to the same strategy");
+
+        if (address(strategy) != address(0)) {
+            strategy.withdrawAll();
+        }
+        strategy = newStrategy;
```

Comment:
Let's get overprotective and call withdrawAll as last thing in the function. Do something like 
```
ITrueStrategy previousStrategy = strategy
strategy = newStrategy
```